agent-codeless/src/main/java/com/microsoft/applicationinsights/agentc/internal/model/TraceContextCorrelationCore.java

```diff
@@ -65,6 +65,38 @@ private TraceContextCorrelationCore() {
         }
     }
 
+    @Nullable
+    public static <Req> DistributedTraceContext resolveCorrelationForInvisibleRequest(Req request,
+                                                                                      Getter<Req> requestHeaderGetter,
+                                                                                      RequestTelemetry requestTelemetry) {
+        try {
+            Traceparent incomingTraceparent = extractIncomingTraceparent(request, requestHeaderGetter);
+            if (incomingTraceparent == null) {
+                return null;
+            }
+
+            // id of the current request
+            requestTelemetry
+                    .setId("|" + incomingTraceparent.getTraceId() + "." + incomingTraceparent.getSpanId() + ".");
+
+            // trace-id of the distributed trace
+            requestTelemetry.getContext().getOperation().setId(incomingTraceparent.getTraceId());
+
+            // assign parent id
+            requestTelemetry.getContext().getOperation().setParentId("|" + incomingTraceparent.getTraceId() + "." +
+                    incomingTraceparent.getSpanId() + ".");
+
+            Tracestate tracestate = getTracestate(request, requestHeaderGetter, incomingTraceparent, getAppId());
+            int traceflag = incomingTraceparent.getTraceFlags();
+
+            return new DistributedTraceContext(requestTelemetry, tracestate, traceflag);
+
+        } catch (Exception e) {
+            logger.debug("unable to perform correlation", e);
```

Comment:
Is this a common occurrence? My first instinct is to log this as a warning.

Reply:
I think the reason it's logged as `debug` (copied from method above `resolveCorrelationForRequest`) is that it can be caused by user sending bad (e.g. malicious) http header and flooding the log